recipes/emsdk/all/conanfile.py

```diff
@@ -28,8 +30,12 @@ def _settings_build(self):
     def layout(self):
         basic_layout(self, src_folder="src")
 
-    def requirements(self):
-        self.requires("nodejs/16.3.0")
+
+    def build_requirements(self):
+        if Version(self.version) < "3.1.68":
+            self.tool_requires("nodejs/16.3.0")
+        else:
+            self.tool_requires("nodejs/20.16.0")
```

Comment:
Where does this restriction come from? Is it LTS releated?

Reply:
Thanks @toge - looking at that thread, as well as:
https://github.com/emscripten-core/emsdk/pull/1476
and the problems that it caused at https://github.com/emscripten-core/emsdk/issues/1485

I think this is not fully related and is orthogonal
The discussions at emsdk pertain to the version of node that ships with _their_ binaries 
from https://github.com/emscripten-core/emsdk/issues/1485#issuecomment-2456057633 - it looks like emscripten itself still supports older nodejs - and the change in their CI was not strictly required and could be reverted: https://github.com/emscripten-core/emsdk/issues/1485#issuecomment-2456055443


As the comments here https://github.com/emscripten-core/emscripten/blob/3.1.72/src/settings.js#L1909-L1914 - there are two aspects:
* the node version to target for the generated code, which is 16.0 for 3.1.72 (the version being added in this PR) see (see https://github.com/emscripten-core/emscripten/blob/3.1.72/src/settings.js#L1909-L1914)
* the minimum version required to run the emscriptem compiler (which I'm not sure where it is specified - it's worth figuring this out).

>  we should clarify if it's really needed or if we could get away with tool_requireing it.

> The issue is that users of emsdk are not able to override nodejs, and they get conflicts using it as a tool_requires


@AbrilRBS  - if `nodejs` is required to "run" the emscriptem compiler from the emsdk package, there's a chance that this needs to a remain a regular requirement and not exclusively a tool_requires - I would reiterate that before changing this, we need to confirm what is the case - do we have a reported github issue for  the override/conflicts ?